### ISSUE TITLE
Enable, Add or Disable headers that are responsible to parse the client's IP

### ISSUE THREAD

**Issue by hiveminded** [get-ion/issues-v1#5]
> _Friday 07 July 2017:v1.0.1_

Devs are now able to modify the allowed header names that are responsible to parse the client's IP from the `Context`.

i.e you want to catch the ip behind a proxy, _X-Real-IP_ and _X-Forward-For_ (these can be modified by user too). 

```go
app.Run([...], [...], ion.WithRemoteAddrHeader("X-Real-IP"), ion.WithRemoteAddrHeader("X-Forwarded-For"))
```

**Outline**:

```go
// WithRemoteAddrHeader enables or adds a new or existing request header name
// that can be used to validate the client's real IP.
//
// Existing values are:
// "X-Real-Ip":             false,
// "X-Forwarded-For":       false,
// "CF-Connecting-IP": false
//
// Look `context.RemoteAddr()` for more.
WithRemoteAddrHeader(headerName string) Configurator

// WithoutRemoteAddrHeader disables an existing request header name
// that can be used to validate the client's real IP.
//
// Existing values are:
// "X-Real-Ip":             false,
// "X-Forwarded-For":       false,
// "CF-Connecting-IP": false
//
// Look `context.RemoteAddr()` for more.
func WithoutRemoteAddrHeader(headerName string) Configurator
```

**Comment by Dexus** [comment]
https://github.com/go-siris/siris/issues/32
@hiveminded please note the use of "CF-Connecting-IP" not "HTTP_CF_CONNECTING_IP"

**Comment by rebootcode** [comment]
@Dexus , Yes Thanks for that. I copied from Cloudflare site earlier (`HTTP_CF_CONNECTING_IP`) but while it was not working, I checked again and updated in my app day before yesterday with `CF-Connecting-IP` while uploading to real server and testing it. I was about to comment here. Though i still found some issue some time, but it is probably my DNS issue. But this is what it is, what you have mentioned.

Thanks for pointing out before me :)
